File: F/1394-find-lucky-integer-in-an-array/main.go
package main

import "fmt"

func findLucky(arr []int) int {
	buf := map[int]int{}
	for i := 0; i < len(arr); i++ {
		buf[arr[i]]++
	}
	ret := -1
	for k, v := range buf {
		if k == v && k > ret {
			ret = k
		}
	}
	return ret
}

func main() {
	fmt.Println("a")
}

File: F/1394-find-lucky-integer-in-an-array/main.rs
use std::collections::HashMap;
impl Solution {
  pub fn find_lucky(arr: Vec<i32>) -> i32 {
    let mut m : HashMap<i32,i32>  = HashMap::new();

    arr.iter().for_each(|&v| {
      m.entry(v).and_modify(|x| *x+=1).or_insert(1);
    });
    let mut ans : i32 = -1;
    for (k, v) in m {
      if v == k {
        ans = ans.max(k);
      }
    }
    ans
  }
}
